### ISSUE TITLE
Stored field values in index?

### ISSUE THREAD

**Issue by edsu** [nextapps-de/flexsearch#90]
I was hoping to reduce the footprint of my index by not storing the original values. I tried the 'memory' preset but I still see the large blocks of text I indexed in the output generated by export(). Is there any way to tell FlexSearch not to store the original values. 

Also, is there documentation available for the options that FlexSearch.create takes? I'm not understanding things like `depth`, `threshold`, `encode`, etc and was wondering if there is any discussion of them somewhere.

Thanks for an awesome JavaScript library!

**Comment by edsu** [comment]
Perhaps a concrete example would help. I'm indexing some multi-page documents, where each page turns into a FlexSearch document:

```js
{
  id: "1234",
  docId: "5678",
  text: "..."
}
```

I created my index like this:

```
const index = new FlexSearch({
  doc: {
    id: "id",
    field: {
      docId: "memory",
      text: "memory"
    }
  }
})
```

After I indexed a little over 2000 docs I saved it with `export()` which produced a 29M file. Then I wrote a small program to remove all the `text` key value pairs in the dictionary that is stored at position 3 in the index;

```python
#!/usr/bin/env python3

import json

i = json.load(open('index.json'))
for key, doc in i[2].items():
    del doc['text']

json.dump(i, open('new-index.json', 'w'))
```

The new file is 26M, so removing the stored text (which I don't need) shaves 3M off of the file size. And, I notice that I can still use it for searching just fine!

So, I was wondering if there is an option to tell FlexSearch not to store particular values in the resulting index. Or I suppose I could post-process things myself, which would be a bit tedious, but at least it's doable.

**Comment by ianloubser** [comment]
@edsu do you specifically require field search functionality? or do you just need full text search ?

**Comment by ts-thomas** [comment]
It is possible to just index document fields and do not store them. You can archive this by using:

```js
// create index
const index = new FlexSearch("memory");
```
```js
// add to index
for(let i = 0; i < docs.length; i++){
    index.add(docs[i].id, docs[i].field);
}
```
```js
// query index
const result = index.search("foo");
```
```js
// enrich results
const doc = docs[result[0]];
```

When you need multiple fields indexed just do:

```js
// create index
const fields = {
    field_1: new FlexSearch("memory"),
    field_2: new FlexSearch("memory"),
    field_3: new FlexSearch("memory")
};
```
```js
// add to index
for(let i = 0; i < docs.length; i++){
    fields.field_1.add(docs[i].id, docs[i].field_1);
    fields.field_2.add(docs[i].id, docs[i].field_2);
    fields.field_3.add(docs[i].id, docs[i].field_3);
}
```
```js
// query index
const result = fields.field_2.search("foo");
```
```js
// enrich results
const doc = docs[result[0]];
```

**Comment by ts-thomas** [comment]
The next update will provide a new field "store" in the document descriptor which separates storing from indexing.

**Comment by edsu** [comment]
So is this new field "store" available yet?

**Comment by ts-thomas** [comment]
Yes, a good time already: https://github.com/nextapps-de/flexsearch#configure-store

**Comment by edsu** [comment]
Is this only available in 0.7.0? I tried it in 0.6.32 and it seemed to have no effect. Here's how I used it, am I doing something wrong?

```javascript
    this.index = new FlexSearch({
      doc: {
        id: "id",
        field: {
          title: {
            encode: "extra",
            tokenize: "strict",
            threshold: 0,
            resolution: 1,
            filter: stopwords
          },
          text: {
            encode: "extra",
            tokenize: "strict",
            threshold: 0,
            resolution: 1,
            filter: stopwords
          },
          description: {
            encode: "extra",
            tokenize: "strict",
            threshold: 0,
            resolution: 1,
            filter: stopwords
          },
          series: {
            encode: "extra",
            tokenize: "strict",
            threshold: 0,
            resolution: 1,
            filter: stopwords
          }
        },
        store: "title"
      }
  })
```

**Comment by edsu** [comment]
I take it back, it is working! Thanks for your help @ts-thomas
